### PR TITLE
Update collaborator endpoints

### PR DESCRIPTION
**Pull request by justrudd**

### CODE REVIEW

File: endpoints/delete_collaborators.md [hunk 1]
```diff
@@ -1 +1,112 @@
...
+[
+	{
+		"account_id": "<string>",
+		"id": "<string>"
+	}
+]
+
+```
+
+At least 1 object with the field `account_id` is required. Once the query is constructed, it is passed in the query string parameter `query`. It must be URL encoded.
```
Comment (ghost) [inline endpoints/delete_collaborators.md:15]: Doesn't the DELETE verb put the data in the body of the request?

File: endpoints/post_collaborators.md [hunk 2]
```diff
@@ -1 +1,122 @@
...
+* `role` is one of the following values - `admin`, `owner`, or `editor`. Please see [here](collaborator_roles.md) for details.
+* `website_ids` is an array of website unique identifiers. This field is only provided if the `role` is `editor`.
+
+At least 1 object is required in the request.
+
+The response is an array of objects. If the collaborator can be created successfully, the object will have the following shape:
+
+```json
+{
+	"_idx": <integer>,
```
Comment (ghost) [inline endpoints/post_collaborators.md:29]: technically this should be <number>

**Review by ghost — CHANGES_REQUESTED** [review]
couple of minor things

File: endpoints/post_collaborators.md (hunk 2 above)
Comment (justrudd) [reply endpoints/post_collaborators.md:29]: Should be? Your comment kind of cuts off...

Comment (justrudd) [reply endpoints/post_collaborators.md:29]: `<Integer>`? Uppercase I?

File: endpoints/delete_collaborators.md (hunk 1 above)
Comment (justrudd) [reply endpoints/delete_collaborators.md:15]: It's ambiguous. The spec doesn't allow nor forbid it. Supposedly the next version of the spec will have this text in it:

> A client SHOULD NOT generate a body in a DELETE request. A payload received in a DELETE request has no defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request.

And I have run across HTTP libraries in the past that don't allow `DELETE` to have a body.

Comment (josh-termly) [reply endpoints/delete_collaborators.md:15]: which why do we want to go? If we do want them in the query params then Steve and I will need to update our PRs

File: endpoints/post_collaborators.md [hunk 3]
```diff
@@ -1 +1,122 @@
...
+* `account_id` is the unique identifier of the account which owns the collaborator.
+* `email` is the email address of the collaborator
+* `first_name` is the Given name of the collaborator if provided by the collaborator upon signing in for the first time.
+* `last_name` is the Surname of the collaborator if provided by the collaborator upon signing in for the first time.
+* `invitation_url` is the URL sent to the collaborator so they can activate their account. If the inivitation has been accepted, this field is null.
+* `invitation_status` indicates whether or not the invite has been accepted.
+* `role` is one of the following values - `admin`, `owner`, or `editor`. Please see [here](collaborator_roles.md) for details.
+* `website_ids` is an array of website unique identifiers. This field is only provided if the `role` is `editor`.
+
+If one of the collaborators cannot be created, the object will be an [error object](../error_object.md). If the error is a validation error, there will be a field called [validation errors](../validation_error_object.md).
```
Comment (josh-termly) [inline endpoints/post_collaborators.md:55]: When I filled in the error details I put them all in the `error_object.md` if we want to operate them into separate files let me know.

File: endpoints/put_collaborators.md [hunk 4]
```diff
@@ -1 +1,91 @@
...
+		"website_ids": [
+			"<string>"
+		]
+	}
+]
+```
+
+* `account_id` is the unique identifier of the account which owns the collaborator.
+* `id` is the unique identifier for the collaborator to update.
+* `role` is one of the following values - `admin`, `owner`, or `editor`. Please see [here](collaborator_roles.md) for details.
```
Comment (josh-termly) [inline endpoints/put_collaborators.md:20]: will there be another PR for `collaborator_roles.md`?

File: endpoints/post_collaborators.md (hunk 2 above)
Comment (ghost) [reply endpoints/post_collaborators.md:29]: never mind, JSON adds an  integer type over JavaScript's number type.

https://json-schema.org/understanding-json-schema/reference/numeric.html

File: endpoints/delete_collaborators.md (hunk 1 above)
Comment (justrudd) [reply endpoints/delete_collaborators.md:15]: @josh-termly query string. The spec is ambiguous so we go with what we know works.

File: endpoints/post_collaborators.md (hunk 3 above)
Comment (justrudd) [reply endpoints/post_collaborators.md:55]: @josh-termly - one file is fine. Once your PR with the definitions get merged, I'll update this.

File: endpoints/put_collaborators.md (hunk 4 above)
Comment (justrudd) [reply endpoints/put_collaborators.md:20]: Yes. I'll do that one later. It's not critical path for getting them out the door

**Pull request merged**